app/controllers/qa/linked_data_terms_controller.rb

```diff
@@ -14,54 +16,83 @@ def index
     head :not_found
   end
 
+  # Return a list of supported authority names
+  # get "/list/linked_data/authorities"
+  # @see Qa::Authorities::LinkedData::AuthorityService#authority_names
+  def list
+    render json: Qa::Authorities::LinkedData::AuthorityService.authority_names.to_json
+  end
+
+  # Reload authority configurations
+  # get "/reload/linked_data/authorities?auth_token=YOUR_AUTH_TOKEN_DEFINED_HERE"
+  # @see Qa::Authorities::LinkedData::AuthorityService#load_authorities
+  def reload
+    Qa::Authorities::LinkedData::AuthorityService.load_authorities
+    list
+  end
+
   # Return a list of terms based on a query
+  # get "/search/linked_data/:vocab(/:subauthority)"
   # @see Qa::Authorities::LinkedData::SearchQuery#search
-  def search # rubocop:disable Metrics/MethodLength
-    begin
-      terms = @authority.search(query, subauth: subauthority, language: language, replacements: replacement_params)
-    rescue Qa::ServiceUnavailable
-      logger.warn "Service Unavailable - Search query #{query} unsuccessful for#{subauth_warn_msg} authority #{vocab_param}"
-      head :service_unavailable
-      return
-    rescue Qa::ServiceError
-      logger.warn "Internal Server Error - Search query #{query} unsuccessful for#{subauth_warn_msg} authority #{vocab_param}"
-      head :internal_server_error
-      return
-    rescue RDF::FormatError
-      logger.warn "RDF Format Error - Results from search query #{query} for#{subauth_warn_msg} authority #{vocab_param} " \
-                  "was not identified as a valid RDF format.  You may need to include the linkeddata gem."
-      head :internal_server_error
-      return
-    end
+  def search
+    terms = @authority.search(query, subauth: subauthority, language: language, replacements: replacement_params)
```

Comment:
Changes to this method are not functional changes.  The method operates exactly the same.  The refactor makes the code cleaner and easier to read.